### DISCUSSION TITLE
Sticky header for double row headers

### DISCUSSION THREAD

**Discussion by ingegneri-bruno-clorox** [KevinVandy/material-react-table#906] [Q&A] [answered]
Hi fellows! 

I got the following scenario where I need a double row header to group metrics of a period.
![Screenshot 2024-01-03 at 13 02 09](https://github.com/KevinVandy/material-react-table/assets/116366661/bdf07fba-c64c-4b13-957d-afc8958cb67e)

But, whenever I do this, I cannot properly make use of the **StickyHeader** functionality. Only the second header row is working fine and being sticky as I scroll down the table, but the first row goes behind it, as you can see it on the above screenshots.
![Screenshot 2024-01-03 at 13 02 15](https://github.com/KevinVandy/material-react-table/assets/116366661/2e981513-779e-4454-a4ac-5909a587de62)

![Screenshot 2024-01-03 at 13 02 27](https://github.com/KevinVandy/material-react-table/assets/116366661/23b5bf63-1bf3-477d-9a33-56c2b3d97ca4)

I have `enableStickyHeader: true` on table props.

My columns got another set of columns inside it, something like that:

```
{
  accessorKey: "values",
  header: $monthYear,
  size: "auto",
  columns: [
    {
      accessorKey: "Xdata"
      size: "auto",
      header: "X"
    },
    {
      accessorKey: "Ydata"
      size: "auto",
      header: "Y"
    },
    {
      accessorKey: "Zdata"
      size: "auto",
      header: "Z"
    }
  ]
};
```

**How can I get this double row header being sticky integrally?**

**Comment by rajaththomson** [comment]
The z-index property might affect the stacking of the elements. Ensure that the first header row has a higher z-index than the content rows so that it does not go behind them as you scroll.

Also verify that the table layout is set to fixed. You can do this by setting the 'table-layout' CSS property of the table to 'fixed'.

**Reply by ingegneri-bruno-clorox** [reply, marked as answer]
Thanks for the reply! That gave me a better perspective of the roots of the issue.
I was not able to work this out with `table-layout` CSS property.

What I had to do was apply these _tableProps_:
```
muiTableHeadProps: {
      sx:{
        position: "sticky",
        top: "0px",
        zIndex: 10,
      }
    }
```
This way, my `<thead>` element stays on top of `<tbody>` cells.

Helped me a lot! Thanks again!
